discovery-infra/day2.py

```diff
@@ -290,3 +304,20 @@ def set_cluster_proxy(client, cluster_id, copy_proxy_from_cluster, args):
     https_proxy = args.https_proxy if args.https_proxy else copy_proxy_from_cluster.https_proxy
     no_proxy = args.no_proxy if args.no_proxy else copy_proxy_from_cluster.no_proxy
     client.set_cluster_proxy(cluster_id, http_proxy, https_proxy, no_proxy)
+
+def set_nodes_hostnames_if_needed(client,
```

Comment:
I don't like the `if_needed` part :( This name exposes the internal logic, which might change. Bad practice IMO.

Reply:
i am open to suggestions. but it explains to the reader that the hostnames are set only in case it is needed